src/protocols/dhcpv4/encode.c

```diff
@@ -73,28 +77,26 @@ static ssize_t encode_value(uint8_t *out, size_t outlen,
 	case FR_TYPE_ETHERNET:
 	case FR_TYPE_STRING:
 	case FR_TYPE_OCTETS:
-		len = fr_value_box_to_network(&need, p, outlen, &vp->data);
+		len = fr_value_box_to_network_dbuff(&need, dbuff, &vp->data);
 		if (len < 0) return -2;
 		if (need > 0) return -1;
-		p += len;
 		break;
 
 	case FR_TYPE_IPV6_PREFIX:
-		if (outlen < (1 + sizeof(vp->vp_ipv6addr))) {
+		len = 1 + sizeof(vp->vp_ipv6addr);
+		if (fr_dbuff_remaining(dbuff) < (size_t)len) {
 			return -1;
 		}
-		*(p++) = vp->vp_ip.prefix;
-		memcpy(p, &vp->vp_ipv6addr, sizeof(vp->vp_ipv6addr));
-		p += sizeof(vp->vp_ipv6addr);
+		fr_dbuff_bytes_in(dbuff, vp->vp_ip.prefix);
+		fr_dbuff_memcpy_in(dbuff, (uint8_t const *)&vp->vp_ipv6addr, sizeof(vp->vp_ipv6addr));
 		break;
 
 	case FR_TYPE_IPV6_ADDR:
-		if (outlen < sizeof(vp->vp_ipv6addr)) {
+		len = sizeof(vp->vp_ipv6addr);
+		if (fr_dbuff_remaining(dbuff) < (size_t)len) {
```

Comment:
This is redundant.  As are the other manual length checks.  You should use the macros which cause the function to return if we're out of space.

Reply:
Thank you; I'll eliminate those in this and in dhcpv6.

Reply:
They're changed to use the macros.